File: examples/python/testing-patterns/complex-workflows/test-conditional-branching.py
@pytest.mark.durable_execution(handler=handler, lambda_function_name="conditional_workflow")
def test_high_value_path(durable_runner):
    """Test high-value orders require approval."""
    with durable_runner:
        result = durable_runner.run(input={"amount": 1500}, timeout=30)
    
    assert result.status is InvocationStatus.SUCCEEDED
    assert deserialize_operation_payload(result.result) == "High-value order processed"
    
    # Verify approval step exists
    approval_step = result.get_step("approval")
    assert approval_step is not None

@pytest.mark.durable_execution(handler=handler, lambda_function_name="conditional_workflow")
def test_standard_path(durable_runner):
    """Test standard orders skip approval."""
    with durable_runner:
        result = durable_runner.run(input={"amount": 500}, timeout=30)
    
    assert result.status is InvocationStatus.SUCCEEDED
    
    # Verify no approval step
    step_names = [op.name for op in result.operations if op.operation_type == OperationType.STEP]
    assert "approval" not in step_names

File: examples/java/testing-patterns/complex-workflows/test-conditional-branching.java
@Test
void testHighValuePath() {
    var runner = LocalDurableTestRunner.create(Map.class, new ConditionalHandler());

    var result = runner.runUntilComplete(Map.of("amount", 1500));

    assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());
    assertEquals("High-value order processed", result.getResult(String.class));

    // Verify approval step exists
    var approvalOp = result.getOperation("approval");
    assertNotNull(approvalOp);
}

@Test
void testStandardPath() {
    var runner = LocalDurableTestRunner.create(Map.class, new ConditionalHandler());

    var result = runner.runUntilComplete(Map.of("amount", 500));

    assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());

    // Verify no approval step
    var approvalOp = result.getOperation("approval");
    assertNull(approvalOp);
}
